src/commands/run.rs

```diff
@@ -475,7 +459,22 @@ fn populate_with_wasi(
             builder = builder.preopened_dir(dir, name)?;
         }
 
-        store.data_mut().wasi = Some(builder.build());
+        host.wasi = Some(builder.build());
+    }
+
+    if wasi_modules.wasi_crypto {
+        #[cfg(not(feature = "wasi-crypto"))]
+        {
+            bail!("Cannot enable wasi-crypto when the binary is not compiled with this feature.");
+        }
+        #[cfg(feature = "wasi-crypto")]
+        {
+            wasmtime_wasi_crypto::add_to_linker(linker, |host| {
+                std::sync::Arc::get_mut(host.wasi_crypto.as_mut().unwrap())
+                    .expect("wasi-crypto is not implemented with multi-threading support")
```

Comment:
Could the `Arc` type be imported into this module? (it's used in a number of places and should avoid the `std::sync::...` prefix)

Additionally can you leave some comments here for what's going on? Basically something along the lines of how `wasi-crypto` the implementation is not ready for threads yet so the `get_mut` implementation here will succeed only if a thread hasn't been spawned. If a thread is spawned, however, this will panic and indicate to the user that work needs to happen to integrate wasi-crypto and threads.

(and then a comment on `wasi-nn` below referring to this comment as well)